### ISSUE TITLE
Phase 6.3: Train World Knowledge Probe (~50M params)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#268]
## Parent Issue
Part of Epic #252: Enhanced AST-Aware Retriever Implementation

## Overview
Train a probe to bridge world knowledge gaps that cannot be expressed in the AST.

## Use Cases
1. **Entity type inference**: "Einstein" → physicist, scientist, person
2. **Temporal reasoning**: "19a jarcento" → 1800-1899
3. **Geographic relations**: "Varsovio" → Poland, Europe
4. **Domain classification**: Recognize scientific, literary, historical topics

## Model Architecture
```
Input: entity_root → Entity embedding → Multi-head probe → Property predictions

Probes:
- Entity type: 20 classes (person, place, organization, ...)
- Domain: 10 classes (science, history, literature, ...)
- Era: 5 classes (ancient, medieval, modern, contemporary, ...)
```

## Training Data
Source: Wikipedia infoboxes, Wikidata, ReVo definitions

```python
def extract_entity_properties(wikipedia_dump):
    """Extract entity properties from Wikipedia."""
    for article in wikipedia_dump:
        entity = article.title
        properties = {
            'type': infer_type_from_categories(article.categories),
            'domain': infer_domain_from_content(article.text),
            'era': extract_temporal_info(article.infobox),
        }
        yield entity, properties
```

## Training Configuration
```yaml
model:
  entity_embedding_dim: 256
  probe_hidden: 512
  num_type_classes: 20
  num_domain_classes: 10
  num_era_classes: 5

training:
  batch_size: 128
  learning_rate: 1e-4
  epochs: 100
  
loss: MultiTaskCrossEntropy
```

## Integration
```python
def enhance_retrieval_with_world_knowledge(query, candidates):
    """Use world knowledge probe to rerank candidates."""
    query_entities = extract_entities(query)
    entity_types = probe.predict_types(query_entities)
    
    for doc in candidates:
        doc_entities = extract_entities(doc)
        doc_types = probe.predict_types(doc_entities)
        
        # Boost if entity types align with question type
        if question_type == WHO and 'person' in doc_types:
            doc.score += 0.5
```

## Success Criteria
- [ ] Entity type accuracy >85%
- [ ] Domain classification accuracy >80%
- [ ] Model size ~50M parameters
- [ ] Saved to models/world_knowledge_probe/
